### PR TITLE
Doc sync behavior in `copy_ptr_to_host`

### PR DESCRIPTION
**Pull request by jakirkham**
Add note about syncing on default stream when no stream is provided.

### CODE REVIEW

File: python/rmm/_lib/device_buffer.pyx [hunk 1]
```diff
@@ -211,6 +211,13 @@ cpdef void copy_ptr_to_host(uintptr_t db,
     hb : ``bytes``-like buffer to write into
     stream : CUDA stream to use for copying, default 0
 
+    Note
+    ----
+
+    If ``stream`` is the default stream, we synchronize on it.
```
Comment (harrism) [inline python/rmm/_lib/device_buffer.pyx]: Please don't use the royal we in docs. The queen doesn't synchronize, this function does.  Also please be explicit about *when* synchronization is done (before the copy? after? both?)

Comment (jakirkham) [reply python/rmm/_lib/device_buffer.pyx]: Ok have made some changes in recent commits. Please take a look 🙂

File: python/rmm/_lib/device_buffer.pyx [hunk 2]
```diff
@@ -214,8 +214,8 @@ cpdef void copy_ptr_to_host(uintptr_t db,
     Note
     ----
 
-    If ``stream`` is the default stream, we synchronize on it.
-    However if a non-default ``stream`` is provided, we leave
+    If ``stream`` is the default stream, it is synchronized after the copy.
+    However if a non-default ``stream`` is provided, leave
```
Comment (harrism) [inline python/rmm/_lib/device_buffer.pyx]:
```suggestion
    However if a non-default ``stream`` is provided, it is up to the user to synchronize.
```

Comment (jakirkham) [reply python/rmm/_lib/device_buffer.pyx]: Not sure why GitHub wouldn't let me commit this as-is. Anyways copied it and committed it in the latest change. Please take a look 🙂

**Pull request merged**